dashboard/app/views/levels/_dialog.html.haml

```diff
@@ -125,20 +125,57 @@
     adjustedScroll = true;
   }
 
+  /**
+   * Allow the app to perform additional actions after the submit button is clicked but
+   * before the dashboard submit button handler code runs.
+   */
+  $(window).on('submit_button_pressed', function(event, onBeforeSubmit, submitButtonSelector) {
+    // Make sure handleSubmit is not inadvertently be called both by this event handler and
+    // the click event handler below.
+    if (!submitButtonSelector || (submitButtonSelector === '.submitButton')) {
+      throw new Error('must provide a submitButtonSelector other than ' +
+          '".submitButton" when firing submit_button_pressed.');
+    }
+
+    handleSubmit(onBeforeSubmit, submitButtonSelector);
+  });
+
   $('.submitButton').click(function () {
-    var submitButton = $('.submitButton');
+    handleSubmit();
+  });
+
+  /**
+   * Process the solution to the puzzle submitted by the user.
+   * @param {function(callback)} onBeforeSubmit Optional function to call
+   * prior to processing the submit action, which calls the callback
+   * function passed to it once complete.
+   * @param {String} submitButtonSelector Optional jQuery selector for
+   * the button to enable/disable to make sure this handler is only called
+   * once. (default: '.submitButton').
+   */
+  function handleSubmit(onBeforeSubmit, submitButtonSelector) {
+    submitButtonSelector = submitButtonSelector || '.submitButton';
+
+    var submitButton = $(submitButtonSelector);
     if (submitButton.attr('disabled')) {
       return;
     }
+    // Avoid multiple simultaneous submissions.
+    submitButton.attr('disabled', true);
+
+    if (onBeforeSubmit) {
+      onBeforeSubmit(processResults.bind(this, submitButtonSelector));
+    } else {
+      processResults(submitButtonSelector);
+    }
```

Comment:
nope - the whole point is that onBeforeSubmit has to happen before processResults.